### PR TITLE
Add adminRefund to BTC HTLC 

### PR DESCRIPTION
**Pull request by KayBeSee**
I'm not sure why it picked up changes at the end of `packages/htlc/src/network-models/evm/contract-artifacts/Swap.json` etc but this PR has the changes from #99 without the bunk EVM contract merges.

### CODE REVIEW

File: packages/htlc/src/network-models/utxo/utils/decode-script.ts [hunk 1]
```diff
@@ -179,14 +205,21 @@ export function getSwapRedeemScriptDetails<N extends Network>(
...
+          OP_EQUAL2,
+          OP_IF2,
+          OP_ELSE2,
           decompiledTimeLockValue,
           OP_TIMELOCKMETHOD,
+          OP_ENDIF,
           OP_DROP2,
-          OP_DUP2,
-          OP_HASH1602,
+          OP_DUP3,
```
Comment (cavanmflynn) [inline packages/htlc/src/network-models/utxo/utils/decode-script.ts]: Nit: Looks like it was already in place, but appending numbers to the opcode variables is a confusing naming convention. Possible to make it more obvious that each is just another instance of the same opcode?

File: packages/htlc/src/network-models/utxo/utils/generate-script.ts [hunk 2]
```diff
@@ -90,8 +110,15 @@ export function createSwapRedeemScript(
...
       claimerPublicKeyBuffer,
       opcodes.OP_ELSE,
-      nSequenceBuffer,
-      opcodes.OP_CHECKSEQUENCEVERIFY,
+      opcodes.OP_DUP,
+      opcodes.OP_HASH160,
+      refundHashRipemd160Buffer,
+      opcodes.OP_EQUAL,
+      opcodes.OP_IF,
+      opcodes.OP_ELSE,
```
Comment (cavanmflynn) [inline packages/htlc/src/network-models/utxo/utils/generate-script.ts]: Better to use `OP_NOTIF` here?

Comment (KayBeSee) [reply packages/htlc/src/network-models/utxo/utils/generate-script.ts]: Definitely, good call.

File: packages/htlc/src/network-models/utxo/utils/decode-script.ts (hunk 1 above)
Comment (KayBeSee) [reply packages/htlc/src/network-models/utxo/utils/decode-script.ts]: I agree, it's a pretty lame naming convention. I'm not sure how we could do it another way. I'm open to ideas/suggestions though.

**Comment by KayBeSee** [comment]
This can be looked at again.

File: packages/htlc/src/network-models/utxo/utils/decode-script.ts [hunk 3]
```diff
@@ -83,93 +83,115 @@ export function getSwapRedeemScriptDetails<N extends Network>(
...
+      paymentHashRipemd160 = decompiledPaymentHashRipemd160;
+
+      timelock = getTimeLockObjectFromDecompiledOpcode(
+        OP_TIMELOCKMETHOD,
+        decompiledTimeLockValue,
+      );
       break;
 
-    case 17:
+    // no adminRefund included in script
```
Comment (cavanmflynn) [inline packages/htlc/src/network-models/utxo/utils/decode-script.ts]: Nit: Incorrect comment

File: packages/htlc/src/network-models/utxo/utils/fee-estimation.ts [hunk 4]
```diff
@@ -66,6 +66,7 @@ export function estimateFee(
     Buffer.from(redeemScript, 'hex'),
   );
 
-  const vRatio = 4; // A witness byte weighs one weight unit; compared to four non-witness weight units.
+  // KBC-TODO: change this back / reexamine this probably
+  const vRatio = 3; // A witness byte weighs one weight unit; compared to four non-witness weight units.
```
Comment (cavanmflynn) [inline packages/htlc/src/network-models/utxo/utils/fee-estimation.ts]: Switch back

File: packages/htlc/src/network-models/utxo/utils/generate-script.ts [hunk 5]
```diff
@@ -73,14 +73,34 @@ export function createSwapRedeemScript(
...
+    refundHashBuffer = Buffer.from(scriptArgs.refundHash, 'hex');
+    refundHashRipemd160Buffer = crypto.ripemd160(refundHashBuffer);
+  }
+
   const paymentHashRipemd160Buffer = crypto.ripemd160(paymentHashBuffer);
 
-  let swapScript: (number | Buffer)[];
+  // set the correct timelock values/method for swapScript
+  let timelockValue;
+  let timelockMethod;
```
Comment (cavanmflynn) [inline packages/htlc/src/network-models/utxo/utils/generate-script.ts]: Nit: add types for safety

```typescript
let timelockValue: Buffer;
let timelockMethod: number;
```

File: packages/htlc/src/network-models/utxo/utxo-htlc.ts [hunk 6]
```diff
@@ -206,6 +240,7 @@ export class UtxoHtlc<N extends Network> extends BaseHtlc<N> {
     unlock: string,
     privateKey: string,
     isClaim?: boolean,
+    adminRefundPublicKey?: string,
```
Comment (cavanmflynn) [inline packages/htlc/src/network-models/utxo/utxo-htlc.ts]: Missing JSDoc comment

**Pull request merged**